contrib/docker/1-start.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+# This file is part of S3QL.
+#
+# Copyright (c) 2008 Nikolaus Rath <Nikolaus@rath.org>
+#
+# This work can be distributed under the terms of the GNU GPLv3.
+
+function debug_log() {
+  if [ "$VERBOSE" == "true" ]; then
+    echo "$1"
+  fi
+}
+
+mount_s3ql_args=()
+
+if [ -n "$LOG_FILE" ]; then
+  debug_log "LOG_FILE: $LOG_FILE"
+  mount_s3ql_args+=("--log" "$LOG_FILE")
+fi
+
+if [ -n "$CACHE_DIR" ]; then
+  debug_log "CACHE_DIR: $CACHE_DIR"
+  mount_s3ql_args+=("--cachedir" "$CACHE_DIR")
+fi
+
+if [ -n "$BACKEND_OPTIONS" ]; then
+  debug_log "BACKEND_OPTIONS: $BACKEND_OPTIONS"
+  mount_s3ql_args+=("--backend-options" "$BACKEND_OPTIONS")
+fi
+
+if [ -n "$AUTH_FILE" ]; then
+  debug_log "AUTH_FILE: $AUTH_FILE"
+  mount_s3ql_args+=("--authfile" "$AUTH_FILE")
+fi
+
+if [ -n "$COMPRESS" ]; then
+  debug_log "COMPRESS: $COMPRESS"
+  mount_s3ql_args+=("--compress" "$COMPRESS")
+fi
+
+if [ -n "$CACHE_SIZE" ]; then
+  debug_log "CACHE_SIZE: $CACHE_SIZE"
+  mount_s3ql_args+=("--cachesize" "$CACHE_SIZE")
+fi
+
+if [ -n "$MAX_CACHE_ENTRIES" ]; then
+  debug_log "MAX_CACHE_ENTRIES: $MAX_CACHE_ENTRIES"
+  mount_s3ql_args+=("--max-cache-entries" "$MAX_CACHE_ENTRIES")
+fi
+
+if [ "$KEEP_CACHE" == "true" ]; then
+  debug_log "KEEP_CACHE: $KEEP_CACHE"
+  mount_s3ql_args+=("--keep-cache")
+fi
+
+if [ "$ALLOW_OTHER" == "true" ]; then
+  debug_log "ALLOW_OTHER: $ALLOW_OTHER"
+  mount_s3ql_args+=("--allow-other")
+fi
+
+if [ "$ALLOW_ROOT" == "true" ]; then
+  debug_log "ALLOW_ROOT: $ALLOW_ROOT"
+  mount_s3ql_args+=("--allow-root")
+fi
+
+if [ -n "$FS_NAME" ]; then
+  debug_log "FS_NAME: $FS_NAME"
+  mount_s3ql_args+=("--fs-name" "$FS_NAME")
+fi
+
+if [ -n "$METADATA_UPLOAD_INTERVAL" ]; then
+  debug_log "METADATA_UPLOAD_INTERVAL: $METADATA_UPLOAD_INTERVAL"
+  mount_s3ql_args+=("--metadata-upload-interval" "$METADATA_UPLOAD_INTERVAL")
+fi
+
+if [ -n "$UPLOAD_THREADS" ]; then
+  debug_log "UPLOAD_THREADS: $UPLOAD_THREADS"
+  mount_s3ql_args+=("--threads" "$UPLOAD_THREADS")
+fi
+
+if [ "$NFS" == "true" ]; then
+  debug_log "NFS: $NFS"
+  mount_s3ql_args+=("--nfs")
+fi
+
+if [ "$VERBOSE" == "true" ]; then
+  debug_log "VERBOSE: $VERBOSE"
+  mount_s3ql_args+=("--debug")
+fi
+
+if ! (
+  set -x
+  mount.s3ql \
+    "${mount_s3ql_args[@]}" \
+    "$STORAGE_URL" \
+    "$MOUNT_POINT"
+); then
+  echo "Failed mount!"
+  exit 1
+fi
```

Comment:
better use
```bash
exec mount.s3ql \
    "${mount_s3ql_args[@]}" \
    "$STORAGE_URL" \
    "$MOUNT_POINT"
```

uses one less sub-shell and does not hides the exit code of S3QL.

Reply:
that's fair, but the reason for always using a subshell here is to `set -x` before ; that's kind of a personal preference thing but I like knowing that "automated" things actually invoked that I could have done manually

maybe it is a bit overkill though...

Reply:
didn't remove subshell (waiting on your opinion on whether it's important or not to print the executed command?), but ensured the exit code is forwarded

Reply:
I'm no bash guru, but I think adding an `exec` will give you the S3QL exit code (since the sub-shell gets replaced with  mount.s3ql)

```bash
if ! (
  set -x
  exec mount.s3ql \
    "${mount_s3ql_args[@]}" \
    "$STORAGE_URL" \
    "$MOUNT_POINT"
); then
  MOUNT_RETCODE=$?
  echo "Failed mount with code $MOUNT_RETCODE"
  exit $MOUNT_RETCODE
fi
```


---


If #202 gets merged and you can use `--fg` and `--log none` (no need for `$LOG_FILE` anymore, too) then you should get rid of the sub-shell and `exec` S3QL directly.

So `1-start.sh` ends with
```bash
exec mount.s3ql \
    "${mount_s3ql_args[@]}" \
    "$STORAGE_URL" \
    "$MOUNT_POINT"
```

or - when you want to output the command we execute (I have no strong feelings on whether that should be done or not)

```bash
echo mount.s3ql \
    "${mount_s3ql_args[@]}" \
    "$STORAGE_URL" \
    "$MOUNT_POINT"
exec mount.s3ql \
    "${mount_s3ql_args[@]}" \
    "$STORAGE_URL" \
    "$MOUNT_POINT"
```

and `entrypoint.sh` will become
```bash
#!/usr/bin/env bash

# This file is part of S3QL.
#
# Copyright (c) 2008 Nikolaus Rath <Nikolaus@rath.org>
#
# This work can be distributed under the terms of the GNU GPLv3.

VERSION="$(fsck.s3ql --version)"
echo "Starting $VERSION"
echo ""

# (maybe move this to 0-init.sh and add check whether the required environment variables are present in 0-init.sh, too)
echo "Will use storage-url '$STORAGE_URL' and mountpoint '$MOUNT_POINT'"
echo ""

echo "*** Init ***"
if /s3ql/bin/0-init.sh; then
  echo "Init done."
else
  exit $?
fi
echo ""

echo "*** Start ***"
exec /s3ql/bin/1-start.sh
```

`2-shutdown.sh` is unnecessary in that case.

Reply:
That is indeed what I envisioned :slightly_smiling_face: :+1: 

Will watch over for when your PR is merged, and then edit in this direction ; the more I think about it, the more I would almost think of disabling `LOG_FILE` selection altogether, and thus "forcing" usage of stdout for users, as it is indeed more idiomatic when it comes to docker. What do you think of that?

Reply:
> [...] the more I think about it, the more I would almost think of disabling `LOG_FILE` selection altogether, and thus "forcing" usage of stdout for users, as it is indeed more idiomatic when it comes to docker. What do you think of that?

Definitely. No need for extra log files. Just always use `--log none`.

Reply:
Actually ended up being able to remove even more code after Nikolaus' suggestion of extending the arg parser :slightly_smiling_face: :+1:

Reply:
(resolving this issue for now, as we agreed and I made the changes requested, reopen if I misunderstood something)